lib/spack/spack/util/web.py

```diff
@@ -105,14 +106,22 @@ def _spider(url, visited, root, depth, max_depth, raise_on_error):
         root = re.sub('/index.html$', '', root)
 
     try:
+        # Disable certificate verification if required.
+        # https://www.python.org/dev/peps/pep-0476
+        context = ssl._create_unverified_context() \
+            if spack.insecure and sys.version_info >= (2, 7, 9) \
+            else None
+
         # Make a HEAD request first to check the content type.  This lets
         # us ignore tarballs and gigantic files.
         # It would be nice to do this with the HTTP Accept header to avoid
         # one round-trip.  However, most servers seem to ignore the header
         # if you ask for a tarball with Accept: text/html.
         req = Request(url)
         req.get_method = lambda: "HEAD"
-        resp = urlopen(req, timeout=TIMEOUT)
+        resp = urlopen(req, timeout=TIMEOUT) \
+            if context is None \
+            else urlopen(req, timeout=TIMEOUT, context=context)
```

Comment:
Get rid of the conditional and just always pass the context here.  The default value is None:

```console
$ pydoc six.moves.urllib.request.urlopen| cat
Help on function urlopen in six.moves.urllib.request:

six.moves.urllib.request.urlopen = urlopen(url, data=None, timeout=<object object>, 
    cafile=None, capath=None, cadefault=False, context=None)
```